snappi_ixnetwork/ixnetworkapi.py

```diff
@@ -210,12 +211,14 @@ def get_metrics(self, request):
         # Need to change the code style when the choice Enum grows big
         if request.choice == 'port':
             response = self.vport.results(request.port)
-            return self.metrics_response().port_metrics.\
-                deserialize(response)
+            metric_res = self.metrics_response()
+            metric_res.port_metrics.deserialize(response)
+            return metric_res
         if request.choice == 'flow':
             response = self.traffic_item.results(request.flow)
-            return self.metrics_response().flow_metrics.\
-                deserialize(response)
+            metric_res = self.metrics_response()
+            metric_res.flow_metrics.deserialize(response)
+            return metric_res
         if request.choice == 'bgpv4':
             return
```

Comment:
Please raise a user-friendly error for unsupported metric choices (listing down the supported ones)